manifest.go

```diff
@@ -0,0 +1,655 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package iceberg
+
+import (
+	"io"
+	"sync"
+
+	iceio "github.com/apache/iceberg-go/io"
+
+	"github.com/hamba/avro/v2/ocf"
+)
+
+// ManifestContent indicates the type of data inside of the files
+// described by a manifest. This will indicate whether the data files
+// contain active data or deleted rows.
+type ManifestContent int32
+
+const (
+	ManifestContentData    ManifestContent = 0
+	ManifestContentDeletes ManifestContent = 1
+)
+
+type fieldSummary struct {
+	ContainsNull bool    `avro:"contains_null"`
+	ContainsNaN  *bool   `avro:"contains_nan"`
+	LowerBound   *[]byte `avro:"lower_bound"`
+	UpperBound   *[]byte `avro:"upper_bound"`
+}
+
+type manifestFileV1 struct {
+	Path               string          `avro:"manifest_path"`
+	Len                int64           `avro:"manifest_length"`
+	PartitionSpecID    int32           `avro:"partition_spec_id"`
+	Content            ManifestContent `avro:"content"`
+	AddedSnapshotID    *int64          `avro:"added_snapshot_id"`
+	AddedFilesCount    *int32          `avro:"added_data_files_count"`
+	ExistingFilesCount *int32          `avro:"existing_data_files_count"`
+	DeletedFilesCount  *int32          `avro:"deleted_data_files_count"`
+	AddedRowsCount     *int64          `avro:"added_rows_count"`
+	ExistingRowsCount  *int64          `avro:"existing_rows_count"`
+	DeletedRowsCount   *int64          `avro:"deleted_rows_count"`
+	Partitions         *[]fieldSummary `avro:"partitions"`
+	KeyMetadata        []byte          `avro:"key_metadata"`
+}
+
+func (*manifestFileV1) Version() int                       { return 1 }
+func (m *manifestFileV1) FilePath() string                 { return m.Path }
+func (m *manifestFileV1) Length() int64                    { return m.Len }
+func (m *manifestFileV1) PartitionID() int32               { return m.PartitionSpecID }
+func (m *manifestFileV1) ManifestContent() ManifestContent { return m.Content }
```

Comment:
can be removed, because content isn't part of a v1 manifest file

Reply:
The function needs to exist so that it meets the interface, but since `Content` doesn't exist in v1, i'll just hardcode it to return `ManifestContentData`.